java/src/main/java/com/nedap/retail/example/websocket/client/RenosApiListener.java

```diff
@@ -29,13 +29,20 @@ public void onMessage(final String message) {
             } else if (message.contains("event")) {
                 handleEvent(message);
             }
-        } catch (IOException e) {
+        } catch (final MessageParsingException e) {
+            LOG.info("The incoming message could not be parsed.");
+        } catch (final Exception e) {
             LOG.error("There was an error while handling the Renos API message", e);
         }
     }
 
-    private void handleResponse(final String message) throws IOException {
+    private void handleResponse(final String message) throws Exception {
```

Comment:
Declaring a function as `throws Exception` is generally a bad idea, because it makes it impossible for someone using that code to know what kinds of errors may be expected, and how to act appropriately on them.